include/drivers/can.h

```diff
@@ -398,6 +398,207 @@ __subsystem struct can_driver_api {
 
 /** @endcond */
 
+#if defined(CONFIG_CAN_STATS) || defined(__DOXYGEN__)
```

Comment:
Can you move this to a private header file in drivers/can ?

Reply:
As noted elsewhere in this PR, this way of adding stats was agreed to be the defacto standard in https://github.com/zephyrproject-rtos/zephyr/pull/39788#issuecomment-982848434. Please see #42880 for a draft proposal on how the needed boilerplate code could be reduced going forward.

Having this in a public header further benefits out-of-tree CAN drivers.